insight/src/main/java/info/nightscout/androidaps/plugins/pump/insight/LocalInsightPlugin.java

```diff
@@ -1281,130 +1283,194 @@ private void processTotalDailyDoseEvent(TotalDailyDoseEvent event) {
-        temporaryBasal.pumpId = pumpID.id;
-        temporaryBasal.percentRate = event.getAmount();
-        temporaryBasal.isAbsolute = false;
-        temporaryBasal.date = timestamp;
+
+        aapsLogger.debug("XXXX event StartTbr history rec Timestamp: " + timestamp + " date: " + event.getEventYear() + "/" + event.getEventMonth() + "/" + event.getEventDay() + " " + event.getEventHour() + ":" + event.getEventMinute() + ":" + event.getEventSecond() + " Rate: " + event.getAmount() + " Duration(min): " + event.getDuration());
+        TemporaryBasal temporaryBasal = new TemporaryBasal(
+                timestamp,
+                T.mins(event.getDuration()).msecs(),
+                event.getAmount(),
+                false,
+                PumpSync.TemporaryBasalType.NORMAL,
+                timestamp,
+                timestamp + 1000L);             // margin added because on several reeadHistory, timestamp could vary
         temporaryBasals.add(temporaryBasal);
     }
 
-    private void processEndOfTBREvent(String serial, List<TemporaryBasal> temporaryBasals, EndOfTBREvent event) {
+    private void processEndOfTBREvent(List<PumpSync.PumpState.TemporaryBasal> temporaryBasals, EndOfTBREvent event) {
         long timestamp = parseDate(event.getEventYear(), event.getEventMonth(), event.getEventDay(),
                 event.getEventHour(), event.getEventMinute(), event.getEventSecond()) + timeOffset;
-        InsightPumpID pumpID = new InsightPumpID();
-        pumpID.eventID = event.getEventPosition();
-        pumpID.pumpSerial = serial;
-        pumpID.eventType = "EndOfTBR";
-        pumpID.timestamp = timestamp;
-        databaseHelper.createOrUpdate(pumpID);
-        TemporaryBasal temporaryBasal = new TemporaryBasal(getInjector());
-        temporaryBasal.durationInMinutes = 0;
-        temporaryBasal.source = Source.PUMP;
-        temporaryBasal.pumpId = pumpID.id;
-        temporaryBasal.date = timestamp - 1500L;
-        temporaryBasals.add(temporaryBasal);
+        aapsLogger.debug("XXXX event StopTbr history rec Timestamp: " + timestamp + " date: " + event.getEventYear() + "/" + event.getEventMonth() + "/" + event.getEventDay() + " " + event.getEventHour() + ":" + event.getEventMinute() + ":" + event.getEventSecond());
+            PumpSync.PumpState.TemporaryBasal temporaryBasal = new PumpSync.PumpState.TemporaryBasal(
+                    timestamp - 1500L,
+                    0L,
+                    100.0,
+                    false,
+                    PumpSync.TemporaryBasalType.NORMAL,
+                    timestamp - 1500L,
+                    timestamp - 500L);
+            temporaryBasals.add(temporaryBasal);
+    }
+
+    /**
+     * Checks the main screen to determine if TBR on pump matches app state.
+     */
+    private void checkAndResolveTbrMismatch(String serial) throws Exception {
+        // compare with: info.nightscout.androidaps.plugins.PumpDanaR.comm.MsgStatusTempBasal.updateTempBasalInDB()
```

Comment:
This comment says PumpDanaR

Reply:
Yes, I didn't change comment, this function checks and update AAPS if there are inconsistances between TBR in database and TBR running in pump after history synchro.
I copied this function from Combo Driver (himself copied from danaR...).
I kept this function for security but I'm not sure it is still necessary now (with my latest commit, I didn't see any new call of it).

Reply:
Initially the combo copied the logic the DanaR pump used to handle that case. The method and logic has since been removed. I just removed the comment from the combo code, the same can be done here. @DigitalDan1 thanks for pointing this out.